nym-api/src/ecash/api_routes/aggregation_axum.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::ecash::error::EcashError;
+use crate::ecash::state::EcashState;
+use crate::node_status_api::models::AxumResult;
+use crate::v2::AxumAppState;
+use axum::extract::Path;
+use axum::{Json, Router};
+use log::trace;
+use nym_api_requests::ecash::models::{
+    AggregatedCoinIndicesSignatureResponse, AggregatedExpirationDateSignatureResponse,
+};
+use nym_api_requests::ecash::VerificationKeyResponse;
+use nym_ecash_time::{cred_exp_date, EcashTime};
+use nym_validator_client::nym_api::rfc_3339_date;
+use std::sync::Arc;
+use time::Date;
+
+/// routes with globally aggregated keys, signatures, etc.
+pub(crate) fn aggregation_routes(ecash_state: Arc<EcashState>) -> Router<AxumAppState> {
+    Router::new()
+        .route(
+            "/master-verification-key:epoch_id",
+            axum::routing::get({
+                let ecash_state = Arc::clone(&ecash_state);
```

Comment:
I'm confused, why are we explicitly passing the `EcashState` in an `Arc` rather than using the state extractor like you're already doing with other routes?

Reply:
ah. I think I know why, because under some conditions the routes wouldn't exist... 
🤔 
if it would make your life easier (and I think it would make the code nicer), what do you think of managing `Option<EcashState>`
and then simply at the beginning of each request have something along the lines of 
```rust
let Some(ecash_state) = ecash_state else {
  // return an error response, idk a 400 maybe?
}
```

Reply:
sorry, I'm using rocket wording here ; D
by managing I meant putting it inside your `AxumAppState`